### PR TITLE
feat(core): return default evm providers url when evm flag is disabled

### PR DESCRIPTION
**Pull request by lykalabrada**
#### What this PR does / why we need it:
- Return default EVM and ETH RPC provider urls when EVM feature flag is disabled
- Add translations for updated Service Provider screen

#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode\*
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*
- [ ] Added translations\*

### CODE REVIEW

File: mobile-app/app/contexts/CustomServiceProvider.tsx [hunk 1]
```diff
@@ -150,26 +152,28 @@ export function CustomServiceProvider(
...
-      [CustomServiceProviderType.ETHRPC]: ethRpcUrl,
+      [CustomServiceProviderType.ETHRPC]: url,
     }));
-  }, [ethRpcUrl]);
+  }, [ethRpcUrl, isEvmFeatureEnabled]);
 
   const isCustomEvmUrl = useMemo(
-    () => currentUrl.EVM !== defaultEvmUrl,
-    [currentUrl.EVM, defaultEvmUrl],
+    () => isEvmFeatureEnabled && currentUrl.EVM !== defaultEvmUrl,
```
Comment (fullstackninja864) [inline mobile-app/app/contexts/CustomServiceProvider.tsx:171]: can we skip this as above if `isEvmFeatureEnabled` is false then we are setting `evmUrl` to `defaultEvmUrl` only?

Comment (lykalabrada) [reply mobile-app/app/contexts/CustomServiceProvider.tsx]: Nice catch! Yep, tried without this extra condition and still works fine. Updated 👍

File: mobile-app/app/contexts/CustomServiceProvider.tsx [hunk 2]
```diff
@@ -150,26 +152,28 @@ export function CustomServiceProvider(
...
-      [CustomServiceProviderType.ETHRPC]: ethRpcUrl,
+      [CustomServiceProviderType.ETHRPC]: url,
     }));
-  }, [ethRpcUrl]);
+  }, [ethRpcUrl, isEvmFeatureEnabled]);
 
   const isCustomEvmUrl = useMemo(
     () => currentUrl.EVM !== defaultEvmUrl,
-    [currentUrl.EVM, defaultEvmUrl],
+    [currentUrl.EVM, defaultEvmUrl, isEvmFeatureEnabled],
```
Comment (fullstackninja864) [inline mobile-app/app/contexts/CustomServiceProvider.tsx:172]:
```suggestion
    [currentUrl.EVM, defaultEvmUrl],
```

File: mobile-app/app/contexts/CustomServiceProvider.tsx [hunk 3]
```diff
@@ -150,26 +152,28 @@ export function CustomServiceProvider(
...
   );
   const isCustomEthRpcUrl = useMemo(
     () => currentUrl.ETHRPC !== defaultEthRpcUrl,
-    [currentUrl.ETHRPC, defaultEthRpcUrl],
+    [currentUrl.ETHRPC, defaultEthRpcUrl, isEvmFeatureEnabled],
```
Comment (fullstackninja864) [inline mobile-app/app/contexts/CustomServiceProvider.tsx:176]:
```suggestion
    [currentUrl.ETHRPC, defaultEthRpcUrl],
```

**Review by fullstackninja864 — APPROVED** [review]

**Pull request merged**